File: MajorityElement.cpp
#include<iostream>
#include<vector>
#include<unordered_map>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        unordered_map<int, int> map;
        for(int i=0;i<nums.size();i++){
            if(map.find(nums[i])!=map.end()){
                map[nums[i]]++;
            }else{
                map[nums[i]] = 1;
            }
            
            if(map[nums[i]] > nums.size()/2) return nums[i];
        }
        
        return 0;
    }
};

int main(){
    
    int n;
    cin>>n;
    
    vector<int> nums;
    for(int i=0;i<n;i++){
        int num;
        cin>>num;
        nums.push_back(num);
    }
    
    Solution *solution = new Solution();
    cout<<solution->majorityElement(nums);
    
    return 0;
    
}

File: MajorityElement.py
class Solution:
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        dic = {}
        for num in nums:
            if num in dic:
                dic[num] += 1
            else:
                dic[num] = 1
            if dic[num] > len(nums)//2:
                return num
                
if __name__=='__main__':
    nums = list(input())
    solution = Solution()
    print(solution.majorityElement(nums))
